doc/dev/msgr2.rst

```diff
@@ -230,15 +185,13 @@ forms, depending on the AUTH_DONE flags:
 
 * If neither FLAG_SIGNED or FLAG_ENCRYPTED is specified, things are simple::
 
-    stream_id
     frame_len
     tag
     payload
     payload_padding (out to auth block_size)
```

Comment:
This is bugging me.  Do we pad out to a block_size starting from the frame_len, or starting from the payload?  In the encrypted case in particular, it makes sense to do it starting with the payload so that we align with the cipher block_size.

But it means that in the general case if we are doing say 4k writes we can't payloads magically.  Does that matter?  My sense from the conversation we had with the DPDK folks is that we can't do that anyway, since the ethernet frame size is something random 4000 or 8000, with IP and TCP headers and such, so no matter what there will be a scatter/gather list for the TCP stream's content.  (Actually, one of the potential wins, I think, is that in the encrypted case we can feed that sglist to the encryption hardware accelleration so that the decrypted content lands in aligned pages without costing the CPU much of anything...but that might need some additiona alignment fields in the frame header?)